### ISSUE TITLE
Sia-UI-v1.2.1-win32-x64 does not exit Windows 10

### ISSUE THREAD

**Issue by f1nasf1** [NebulousLabs/Sia-UI#588]
Sia-UI-v1.2.1-win32-x64 does not exit in Windows 10 when the top right X button is clicked.  I had to go to the Task Manager and end tasks for all Electron processes which seem to keep starting after I tried to exit the tasks.  At the end, I expanded on one Electron process and saw the Sia UI task and ended that.  Then the UI finally exited.

Am very surprised that no installer was provided for windows 10.

**Comment by Fornax96** [comment]
There should be a Sia icon on your system tray, when you right click it it shows a button to close Sia.

Note that killing Sia processes from the task manager can cause wallet corruption on windows.

**Comment by f1nasf1** [comment]
Unfortunately in Windows 10 the right click and close does not work.  Perhaps I can post a video on github. 

A.

**Comment by bogdanteodoru** [comment]
@avanchev, I just installed latest version of Sia wallet on a brand new Windows 10 and seems to work. Right click works and options are there. Keep in mind that it takes couple of seconds for Sia to close.

Also, if you can create a video that would help the devs a lot (just add to to YouTube)

**Comment by ghost** [comment]
Hey guys, I have the same issue.
I made a quick and dirty [YouTube video](https://www.youtube.com/watch?v=E1O3t56qpck) about it but yeah, it's pretty much as the title said.
Donations are welcome btw ;)

**Comment by bogdanteodoru** [comment]
Hey @gav5999. What you showed in the video with the right click -> close its actually the same functionality as pressing the X button on the Sia UI wallet. Please look in the right corner of the toolbar where the small icons are and see if there's a Sia icon there. If so, right click and "quit sia" should do the trick :)

**Comment by ghost** [comment]
Oh I get it now. If anyone else is also dumbfounded by this. He means [this ](https://i.imgur.com/Xd037Ui.jpg)icon.

**Comment by ghost** [comment]
Oh thanks, I get it now. If anyone else is dumbfounded by this, he means [this ](https://i.imgur.com/Xd037Ui.jpg)icon.

**Comment by bogdanteodoru** [comment]
@moimael or @lukechampine , I think you can close this one.

**Comment by jeremys18** [comment]
There is no way that should be how a program is closed and there is nothing telling users that they have to close it that way. This is an issue and made me think the program was a virus refusing to shut down.

Btw I too force closed it right after installing because it wouldn't close. When I click the x I expect a program to close, not minimize. And when I right click and hit close window I expect it to close not ignore my command. 

Now the question is, are you going to fix it or ignore it?

**Comment by avahowell** [comment]
This is behavior that was specifically implemented because Sia is a long-running network process that should be left open in order to ensure the health of the files you have uploaded and the health of the files other users have uploaded to you. It closely follows the `daemon` design of Sia, and this pattern can be observed in other similar programs that are designed to be run for long periods of time (qBittorrent, flux, Steam, Discord are a few of the ones I'm familiar with.) We could do better with the communication here, but last I checked there isn't an easy way to manage cross-platform system notifications with Electron.

Closing this issue for now

**Comment by jeremys18** [comment]
Actually, this is not how steam behaves. I have steam and when I right click the icon it not only gives me the  "Exit Steam" option, but it exists steam if I choose it. I don't need to go to the tiny area in the corner and hunt for the steam icon, right click, then hit exit. Of course, I can and if I close steam normally it does do that. But as a user I know. It doesn't do what this issue reports.

The issue here is 2 fold.

1. There's no information to the user explaining anything.
2. When close is clicked it minimizes, not just to the tiny icon area, but to the taskbar, which is verrrrrrry confusing to a user.

Both issues can be solved easily though simply by having the UI tell the user in a pop-up when first run, that "hey, you may be wondering soon how to properly close this (like me when I wanted to change the config file/data location). You may think the taskbar is how to close it but naw, you have to close it by right clicking the icon in the system tray instead of the one in the taskbar".

The better approach would be to implement behavior like steam and minimize to the system tray instead of a regular minimize action. That would follow the behavior you outlined above. And of course let the user know it was minimized not closed (most programs do so with an animation and/or balloon message ).

As of now though, you minimize the program to the taskbar which isn't what the user wanted (they tried to close not minimize). Then when the user thinks "oh it's one of those programs" and right clicks the program in the taskbar and select close, you ignore them. That screams unwanted program and/or virus since the program appears to be taking over things (like those annoying popups on fake sites that never let you close the page). This is bad for your program since users can easily report the thing as a virus (which I almost did) and get every virus scanner to remove the files for future users.

**Comment by ghost** [comment]
I fully agree to what Jeremy said above, if it's minimized it'll frustrate anyone that uses it but there also needs to be something to tell the (casual) user on how to completely close it in the taskbar

**Comment by apertamono** [comment]
I had the same misunderstanding and I also used the Task Manager to close Sia-UI. If multiple users are telling you removing a basic functionality of any program is confusing and frustrating, don't tell them they're doing it wrong. Sia-UI is not a daemon, it's a user interface. By the way, I managed to close the daemon by typing 'stop' in the Terminal tab before I found out how to close the UI. Typing 'exit' in the Terminal tab didn't do anything.

**Comment by seandotau** [comment]
Users expect that when X (closed) is clicked, a certain action occurs. Same with "_" (ie minimize). Design 101. Am a fan of Sia but this is amateur.

**Comment by Fornax96** [comment]
A lot of desktop apps actually minimize when you click the X button. Slack and telegram are living examples of this, Chromium does this too by default, but it's configurable. It's expected behaviour that an app that needs to run as a daemon in the background doesn't shut itself down when you click the X button.

Strange thing is that Sia always shuts itself down when I click the X button, the only way to minimize it to the tray is by opening the context menu of the tray icon and selecting "Hide Sia". Is this behaviour different across operating systems, or is it because I'm running a 1.3.0 release candidate? This behaviour is consistent across Linux Mint and Arch Linux with GNOME.

**Comment by MattPark** [comment]
Also -- would be nice if there was an option to close just sia-ui without closing siad.  Especially a remote siad process.

**Comment by lukechampine** [comment]
@MattPark I believe if you start `siad` manually and subsequently start Sia-UI, closing Sia-UI will not shut down `siad`.

**Comment by MattPark** [comment]
@lukechampine Maybe.  I'm doing the remote siad server workaround though.   Where siad.exe on windows is replaced with a script that port forwards localhost 8890 to my remote siad linux server.  

So in this case when I close Sia-UI, it sends shutdown signal over rpc, which shuts down the remote server.
I can block that from happening by manually killing the port forward first.

Just a hassle, I'm sure when you get around to implementing remote serve functionality it will be OBE.
